frontend/src/features/permits/components/permit-list/ApplicationsInReviewRowOptions.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from "react";
+import { OnRouteBCTableRowActions } from "../../../../common/components/table/OnRouteBCTableRowActions";
+import { useUpdateApplicationQueueStatusMutation } from "../../hooks/hooks";
+import { CASE_ACTIVITY_TYPES } from "../../types/CaseActivityType";
+import { ApplicationInReviewModal } from "./ApplicationInReviewModal";
+import { useQueryClient } from "@tanstack/react-query";
+
+const PERMIT_ACTION_OPTION_TYPES = {
+  WITHDRAW_APPLICATION: "withdrawApplication",
+} as const;
+
+type PermitActionOptionType =
+  (typeof PERMIT_ACTION_OPTION_TYPES)[keyof typeof PERMIT_ACTION_OPTION_TYPES];
+
+const getOptionLabel = (optionType: PermitActionOptionType): string => {
+  if (optionType === PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION) {
+    return "Withdraw Application";
+  }
+
+  return "";
+};
+
+const ALL_OPTIONS = [
+  {
+    label: getOptionLabel(PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION),
+    value: PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION,
+  },
+];
+
+const getOptions = (isInReview: boolean) => {
+  return ALL_OPTIONS.filter((option) => {
+    // Exclude 'WITHDRAW_APPLICATION' if 'isInReview' is false
+    if (
+      isInReview &&
+      option.value === PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION
+    ) {
+      return false;
+    }
+    return true;
+  });
+};
+
+export const ApplicationsInReviewRowOptions = ({
+  isInReview,
+  permitId,
+}: {
+  isInReview: boolean;
+  permitId: string;
+}) => {
+  const [isAIRModalOpen, setIsAIRModalOpen] = useState<boolean>(false);
+
+  const queryClient = useQueryClient();
+
+  const handleCloseAIRModal = () => {
+    setIsAIRModalOpen(false);
+    queryClient.invalidateQueries({
```

Comment:
I feel like it might be better to encapsulate this ```queryClient.invalidateQueries``` logic into another method, and move that method inside the file where the other hooks related to applications in review are in (eg. where useUpdateApplicationQueueStatusMutation is in). Then call that method here.

Reply:
@zgong-gov Since this is a single function call which is only made twice (here and in the useUpdateApplicationQueueStatusMutation hook), this approach seems to me like a bit of an unnecessary abstraction which adds complexity. However if you think it best I am happy to do so.

Reply:
@zgong-gov I just tested this approach of encapsulating the invalidateQueries logic into a custom hook call which can be reused in the useUpdateApplicationQueueStatusMutation and handleCloseAIRModal function. This results in a React invalid hook call of "useQueryClient".

`export const useInvalidateApplicationsInQueue = () => {
  const queryClient = useQueryClient();
  queryClient.invalidateQueries({
    queryKey: ["applicationsInQueue"],
  });
};` 

`
  const handleCloseAIRModal = () => {
    setIsAIRModalOpen(false);
    useInvalidateApplicationsInQueue();
  };
  `

Reply:
You can try to do it like this:
```
export const useInvalidateApplicationsInQueue = () => {
    const queryClient = useQueryClient();

    return {
        invalidate: () => {
             queryClient.invalidateQueries({  queryKey: ["applicationsInQueue"]  });
        },
    };
};
```
and inside ApplicationsInReviewRowOptions.tsx:
```
const { invalidate } = useInvalidateApplicationsInQueue();
...
const handleCloseAIRModal = () => {
     setIsAIRModalOpen(false);
     invalidate();
};
```